aws/rust-runtime/aws-types/src/credentials/provider.rs

```diff
@@ -13,54 +13,114 @@ use std::time::Duration;
 #[non_exhaustive]
 pub enum CredentialsError {
     /// No credentials were available for this provider
-    CredentialsNotLoaded,
+    #[non_exhaustive]
+    CredentialsNotLoaded {
+        context: Box<dyn Error + Send + Sync + 'static>,
```

Comment:
Why is this one called `context` while the others are called `cause`?

Reply:
I felt that in the not-loaded case, it was more of a "here's why?" than a "this was the problem that caused this" But I'm very open to "nah, lets just keep it consistent"